src/foam/dao/FUIDAO.js

```diff
@@ -70,12 +65,10 @@ foam.CLASS({
       javaCode: `
         var id = getPropertyInfo().get(obj);
         var klass = getPropertyInfo().getValueClass();
-        synchronized (this) {
-          if ( klass == String.class && ((String) id).isBlank() ) {
-            getPropertyInfo().set(obj, getUIDGenerator().getNextString());
-          } else if ( klass == long.class && ((long) id) == 0L ) {
-            getPropertyInfo().set(obj, getUIDGenerator().getNextLong());
-          }
+        if ( klass == String.class && ((String) id).isBlank() ) {
```

Comment:
why not  ! SafetyUtil.isEmpty  to also test for null? 
or 
```suggestion
        if ( klass == String.class && id != null && ((String) id).isBlank() ) {
```